spec/components/post_destroyer_spec.rb

```diff
@@ -217,6 +217,32 @@
     end
   end
 
+  describe "recovery and public post actions" do
+    let(:codinghorror) { Fabricate(:coding_horror) }
+    let!(:like) { PostAction.act(codinghorror, post, PostActionType.types[:like]) }
+    let!(:another_like) { PostAction.act(moderator, post, PostActionType.types[:like]) }
+
+    it "restores public post actions" do
+      PostDestroyer.new(moderator, post).destroy
+      expect(PostAction.find_by(id: like.id)).not_to be_present
```

Comment:
Since we don't need to fetch the record here, `PostAction.exists?(id: like.id)` is going to use less memory and be faster.